stock_release_channel_shipment_advice_deliver/models/shipment_advice.py

```diff
@@ -0,0 +1,106 @@
+        creation and process them. Each shipment advice manage its own process and call
+        the release channel to notify it when it's done.
+        """
+        return self.release_channel_id and self.release_channel_id.state in (
+            "delivering",
+            "delivering_error",
+        )
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        records = super().create(vals_list)
+        for rec in records:
+            if rec._is_auto_process:
+                rec.with_delay(
+                    description=_(
+                        "Automatically process the shipment advice %(name)s.",
+                        name=rec.name,
+                    )
+                )._auto_process()
+        return records
+
+    def _auto_process(self):
+        self.ensure_one()
+        if not self._is_auto_process:
+            return False
+        if not self.arrival_date:
+            self.arrival_date = fields.Date.context_today(self)
+        self.in_release_channel_auto_process = True
+        try:
+            with self.env.cr.savepoint():
+                move_lines = self.planned_move_ids.move_line_ids
+                move_lines_to_load = move_lines.filtered(
+                    lambda ml: ml.state not in ("done", "cancel")
+                )
+                move_lines_to_load._load_in_shipment(self)
+                if self.state == "confirmed":
+                    self.action_in_progress()
+                self.action_done()
+        except UserError as error:
+            _logger.error(error)
+            self.write(
+                {
+                    "state": "error",
+                    "error_message": self._get_error_message(error, self),
+                }
+            )
+            self.release_channel_id._shipment_advice_auto_process_notify_error(
+                self.error_message
+            )
+        return True
+
+    def _postprocess_action_done(self):
+        res = super()._postprocess_action_done()
+        if self.state == "error":
+            return self.release_channel_id._shipment_advice_auto_process_notify_error(
+                self.error_message
+            )
+        if self.state != "done":
+            return res
+        return self.release_channel_id._shipment_advice_auto_process_notify_success()
+
+    def action_done(self):
+        # If the channel is in error and we try to validate its shipment
+        # advice, we set the release channel state to delivering."""
+        for rec in self:
+            if (
+                rec.state == "error"
+                and rec.release_channel_id
+                and rec.release_channel_id.state == "delivering_error"
+            ):
+                rec.release_channel_id.state = "delivering"
+        return super().action_done()
+
+    def action_in_progress(self):
+        res = super().action_in_progress()
+        self.release_channel_id.filtered(
```

Comment:
Hi @jbaudoux I have a trouble about this
* I create a picking (called P1) with a release channel, called C1 (C1's state is `open`, planning method is `simple`)
* I also create a shipment advice (called SA1) from C1 -> `picking_to_plan_ids` contains P1
* After that I plan picking P1 in shipment SA1 -> `picking_to_plan_ids` will be null because picking is planned
* I lock the SA1 and check auto_deliver
* If I continue to start the SA1 -> nothing happens in C1 because `is_action_deliver_allowed` is calculated based on `picking_to_plan_ids` but it is null now ([here](https://github.com/OCA/wms/pull/819/files#diff-4d3327d6e5d2a44ca6373fd21fd676ce1f565a0c691f73b5b3b09bde0950152fR59)), so C1 cannot change the state

Could you please give me an advice in this one? Maybe I've missed something

Reply:
Hi @jbaudoux the test is [here](https://github.com/jbaudoux/wms/pull/7) please take a look

Reply:
@tuantrantg Is this solved ?